File: rust/src/models/product_family_create_request.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct ProductFamilyCreateRequest {
    pub name: String,
}

impl ProductFamilyCreateRequest {
    pub fn new(name: String) -> Self {
        Self { name }
    }
}

File: typescript/src/models/productFamilyCreateRequest.ts
// this file is @generated

export interface ProductFamilyCreateRequest {
  name: string;
}

export const ProductFamilyCreateRequestSerializer = {
  _fromJsonObject(object: any): ProductFamilyCreateRequest {
    return {
      name: object["name"],
    };
  },

  _toJsonObject(self: ProductFamilyCreateRequest): any {
    return {
      name: self.name,
    };
  },
};
